### PR TITLE
S3t4 check word

### PR DESCRIPTION
**Pull request by daedalus-9**

### CODE REVIEW

File: js/index.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
+function shuffle(array) {
+    let currentIndex = array.length,  randomIndex;
```
Comment (devhenlee) [inline js/index.js]: Hi! There's an extra space before `randomIndex`.

File: js/index.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+            array[randomIndex], array[currentIndex]];
+    }
+
+    return array;
+}
+
+let expectedWord = []
+fetch('words.json')
+    .then(response => response.json())
+    .then(words => {
```
Comment (devhenlee) [inline js/index.js]: Currently, `fetch` returns an error (see below). 

Tried to see whether uploading the `words.json` file onto cloudinary helps, and it did! 
Can you guys upload `words.json` and replace the file path with the cloudinary url?

Error message: 
`Access to fetch at 'file:///Users/henrylee/sites/academyServer/html/2023-mar-wordle/words.json' from origin 'null' has been blocked by CORS policy: Cross origin requests are only supported for protocol schemes: http, data, isolated-app, chrome-extension, chrome, https, chrome-untrusted.`

File: js/index.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+                const result = checkWord(expectedWord,attemptedWord)
+                alert(result)
+            } else {
+                console.log('enter 5 characters')
+            }
+        }
+
+// MAKE THE REAL KEYBOARD WORK
+        const alphabet = 'abcdefghijklmnopqrstuvwxyz'
+        document.addEventListener('keyup', (event) => {
```
Comment (devhenlee) [inline js/index.js]: Can you change the `alphabet` constant to accept capital letters too?

**Comment by devhenlee** [comment]
Also, can you help make fix the row of inputs after users hit enter to submit a guess? Otherwise, they can just press delete to modify their guesses.

File: js/index.js [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+        expectedWord = shuffledWords[0].toUpperCase().split("")
+        console.log(expectedWord)
+        let attemptedWord = []
+        let characterCounter = 0
+// let attemptCounter = 0 - useful later
+
+// MAKE ON SCREEN LETTERS WORK
+        const onScreenLetters = document.querySelectorAll('.key')
+        const backspace = document.querySelector('.backspace')
+        const enter = document.querySelector('.enter')
```
Comment (devhenlee) [inline js/index.js]: Should we rename these constants to e.g. `targetBackspace` to improve: 
- semantics and; 
- consistency (with `instructions.js`)?

File: js/index.js (hunk 1 above)
Comment (daedalus-9) [reply js/index.js]: Amended!

File: js/index.js (hunk 2 above)
Comment (daedalus-9) [reply js/index.js]: Resolved - had to be using localhost

File: js/index.js (hunk 3 above)
Comment (daedalus-9) [reply js/index.js]: Added :)

File: js/index.js [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+        }
+
+        function enterPressed () {
+            if (attemptedWord.length === 5) {
+                const result = checkWord(expectedWord,attemptedWord)
+                alert(result)
+            } else {
+                console.log('enter 5 characters')
+            }
+        }
```
Comment (devhenlee) [inline js/index.js]: Please help remove the `else`; `console.log` runs fine without it. 

Further, do you think we should consider removing the `console.log`? Nothing needs to happen if users press enter before they finish making a guess.

File: js/index.js [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+
+// MAKE THE REAL KEYBOARD WORK
+        const alphabet = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ'
```
Comment (devhenlee) [inline js/index.js]: How about a more semantic constant name such as `characterSet`?

File: js/index.js (hunk 4 above)
Comment (daedalus-9) [reply js/index.js]: amended!

File: js/index.js (hunk 5 above)
Comment (daedalus-9) [reply js/index.js]: Amended!

File: js/index.js [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+                    console.log(attemptedWord, characterCounter)
+                }
+            })
+        })
+
+        backspace.addEventListener('click', deleteLetter)
+        enter.addEventListener('click', enterPressed)
+
+        function deleteLetter (e) {
+            if (characterCounter > 0) {
```
Comment (devhenlee) [inline js/index.js]: The parameter `e` in the `deleteLetter` function isn't used/returned anywhere; okay to remove.

File: js/index.js (hunk 6 above)
Comment (daedalus-9) [reply js/index.js]: Amended!

File: js/index.js [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+            }
+        })
+
+        // console.log(attemptedWord)
+
+        function checkWord(expectedWord,attemptedWord) {
+            let correctLetters = 0
+            let correctPositions = 0
+            for (let i=0; i < attemptedWord.length; i++) {
+                if (attemptedWord[i]===expectedWord[i]) {
```
Comment (devhenlee) [inline js/index.js]: Please help add spaces around the `===` operator to have consistent formatting.

File: js/index.js (hunk 7 above)
Comment (daedalus-9) [reply js/index.js]: Amended!

File: js/index.js [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+                enterPressed()
+            }
+        })
+
+        // console.log(attemptedWord)
+
+        function checkWord(expectedWord,attemptedWord) {
+            let correctLetters = 0
+            let correctPositions = 0
+            for (let i=0; i < attemptedWord.length; i++) {
```
Comment (devhenlee) [inline js/index.js]: Please add spaces for `i=0` as well.

File: js/index.js (hunk 8 above)
Comment (daedalus-9) [reply js/index.js]: Amended!

File: js/index.js (hunk 9 above)
Comment (daedalus-9) [reply js/index.js]: Amended!

File: js/index.js [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+                    correctPositions++
+                }
+                if (expectedWord.includes(attemptedWord[i]) && attemptedWord[i] !== expectedWord[i]){
+                    correctLetters++
+                }
+            }
+            return correctPositions === 5 ? 'Correct' : 'Incorrect'
+        }
+    })
+    .catch(error => console.error(`An error occurred: ${error.message}`))
```
Comment (devhenlee) [inline js/index.js:97]: @ashleycoles Ash:
What do you think about having `.catch`? Do we need it for <em>this</em> project?

**Pull request merged**